runtime/runtime-params-estimator/src/utils.rs

```diff
@@ -283,26 +283,26 @@ pub(crate) fn aggregate_per_block_measurements(
     let mut block_costs = Vec::new();
     let mut total_ext_costs: HashMap<ExtCosts, u64> = HashMap::new();
     let mut total = GasCost::zero();
-    let mut n = 0;
+    let num_measurements = measurements.len() as u64;
     for (gas_cost, ext_cost) in measurements {
         block_costs.push(gas_cost.to_gas() as f64);
         total += gas_cost;
-        n += block_size as u64;
         for (c, v) in ext_cost {
             *total_ext_costs.entry(c).or_default() += v;
         }
     }
     for v in total_ext_costs.values_mut() {
+        let n = num_measurements * block_size as u64;
         *v /= n;
     }
-    let mut gas_cost = total / n;
+    let mut gas_cost = total / num_measurements;
     if is_high_variance(&block_costs) {
         gas_cost.set_uncertain("HIGH-VARIANCE");
     }
     if let Some(overhead) = overhead {
         gas_cost = gas_cost.saturating_sub(&overhead, &NonNegativeTolerance::PER_MILLE);
     }
-    (gas_cost, total_ext_costs)
+    (gas_cost / block_size as u64, total_ext_costs)
```

Comment:
I wonder if we can come up with some naming convention or some such which makes it clearer what is divided by what. It's a bit unnerving that `gas_cost` migrated in meaning from "cost per tx" to "cost per block". 

Perhaps use `block_` for block and `tx_` for transactions? so, `block_cost` and `tx_cost` and `num_blocks` and `num_txes`?

Reply:
Yeah good call, making this code more readable is worth it. I tried some renaming similar to your suggestion and added some comments, hopefully it is clearer now.

I didn't go with `tx` though. Technically it would be correct today, as all caller-sites set `block_size` to the number of transactions. But I think the function signature generally should also allow to define work items as actions, or even single loop iterations within a smart contract method.